### PR TITLE
show list of addons for preview targets

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- Show users basic card for preview postgres
- This is just a start to indicate to users that the database exists - therefore the card won't be clickable until we have a screen for updating the addon after it has been created

### CODE REVIEW

File: api/server/handlers/addons/list.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	DeploymentTargetID string `schema:"deployment_target_id"`
+}
+
+// LatestAddonsResponse represents the response from the /addons/latest endpoint
+type LatestAddonsResponse struct {
+	Base64Addons []string `json:"base64_addons"`
+}
+
+func (c *LatestAddonsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, span := telemetry.NewSpan(r.Context(), "serve-list-addons")
```
Comment (ferozemohideen) [inline api/server/handlers/addons/list.go:46]: span name

File: api/server/handlers/addons/list.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	request := &LatestAddonsRequest{}
+	if ok := c.DecodeAndValidate(w, r, request); !ok {
+		err := telemetry.Error(ctx, span, nil, "error decoding request")
+		c.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(err, http.StatusBadRequest))
+		return
+	}
+
+	telemetry.WithAttributes(span,
+		telemetry.AttributeKV{Key: "project-id", Value: project.ID},
+		telemetry.AttributeKV{Key: "cluster-id", Value: cluster.ID},
```
Comment (ferozemohideen) [inline api/server/handlers/addons/list.go:61]: are these still needed? i forget

File: api/server/handlers/addons/list.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		telemetry.AttributeKV{Key: "deployment-target-id", Value: request.DeploymentTargetID},
+	)
+
+	var deploymentTargetIdentifier *porterv1.DeploymentTargetIdentifier
+	if request.DeploymentTargetID != "" {
+		deploymentTargetIdentifier = &porterv1.DeploymentTargetIdentifier{
+			Id: request.DeploymentTargetID,
+		}
```
Comment (ferozemohideen) [inline api/server/handlers/addons/list.go:69]: can you pass this in even if request.DeploymentTargetID == ""?

File: api/server/handlers/addons/list.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	res := &LatestAddonsResponse{
+		Base64Addons: []string{},
+	}
+
+	for _, addon := range latestAddonsResp.Msg.Addons {
+		by, err := helpers.MarshalContractObject(ctx, addon)
+		if err != nil {
+			err = telemetry.Error(ctx, span, err, "error marshaling addon")
```
Comment (ferozemohideen) [inline api/server/handlers/addons/list.go:98]: maybe log the addon b64 before erroring (if it's not sensitive)

File: api/server/router/addons.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+	newPath := &types.Path{
+		Parent:       basePath,
+		RelativePath: relPath,
+	}
+
+	var routes []*router.Route
+
+	// GET /api/projects/{project_id}/clusters/{cluster_id}/addons/latest -> addons.LatestAddonsHandler
+	latestAddonsEndpoint := factory.NewAPIEndpoint(
```
Comment (ferozemohideen) [inline api/server/router/addons.go:61]: just want to put on the radar that we might want to introduce deployment target scoped routes (/projects/<>/clusters/<>/targets/<>/addons) so we don't have to keep passing it in as a query param

**Review by ferozemohideen — APPROVED** [review]

File: api/server/handlers/addons/list.go (hunk 1 above)
Comment (ferozemohideen) [reply api/server/handlers/addons/list.go:46]: or that might be correct actually, not sure if there's a difference between list all and list latest

File: api/server/router/addons.go (hunk 5 above)
Comment (ianedwards) [reply api/server/router/addons.go:61]: sounds good, should be a pretty easy refactor if needed

File: api/server/handlers/addons/list.go (hunk 3 above)
Comment (ianedwards) [reply api/server/handlers/addons/list.go:69]: it'll be nil if the ID is an empty string, which will cause for the cluster default to be used

File: api/server/handlers/addons/list.go (hunk 1 above)
Comment (ianedwards) [reply api/server/handlers/addons/list.go:46]: Do we have a different pattern here?

File: api/server/handlers/addons/list.go (hunk 2 above)
Comment (ianedwards) [reply api/server/handlers/addons/list.go:61]: probably not, I'll remove

File: api/server/handlers/addons/list.go (hunk 4 above)
Comment (ianedwards) [reply api/server/handlers/addons/list.go:98]: looks like when I do this in a loop then only the last is included, so it could be a bit confusing

**Pull request merged**